src/main/java/io/cdap/cdap/metrics/process/gcp/GoogleCloudMonitoringUtil.java

```diff
@@ -0,0 +1,118 @@
+   *
+   * @param timeSeries Map of {@link TimeSeriesMetadata} and their long values
+   * @param startTime  start timestamp
+   * @param endTime    end timestamp
+   * @return {@link List<TimeSeries>} values
+   */
+  public static List<TimeSeries> convertToMonitoringTimeSeries(Map<TimeSeriesMetadata, List<Long>> timeSeries,
+                                                               long startTime, long endTime) {
+
+    List<TimeSeries> timeSeriesList = new ArrayList<>();
+    for (Map.Entry<TimeSeriesMetadata, List<Long>> entry : timeSeries.entrySet()) {
+      TimeSeriesMetadata metadata = entry.getKey();
+      long value = getAggregateValue(entry, metadata);
+      TimeInterval interval = TimeInterval.newBuilder()
+        .setStartTime(Timestamps.fromSeconds(metadata.getMetricKind() == MetricType.GAUGE ? endTime : startTime))
+        .setEndTime(Timestamps.fromSeconds(endTime))
+        .build();
+
+      TypedValue typedValue = TypedValue.newBuilder().setInt64Value(value).build();
+      Point point = Point.newBuilder().setInterval(interval).setValue(typedValue).build();
+      // Prepares the metric descriptor
+      Metric metric = Metric.newBuilder()
+        .setType(metadata.getMetricType())
+        .putAllLabels(metadata.getMetricLabels())
+        .build();
+      // Prepares the monitored resource descriptor
+      MonitoredResource resource = MonitoredResource.newBuilder()
+        .setType(metadata.getResourceType())
+        .putAllLabels(metadata.getResourceLabels())
+        .build();
+      // Prepares the time series
+      TimeSeries series = TimeSeries.newBuilder()
+        .setMetric(metric)
+        .setResource(resource)
+        .addAllPoints(Collections.singletonList(point))
+        .build();
+      timeSeriesList.add(series);
+    }
+    return timeSeriesList;
+  }
+
+  /**
+   * Send the {@link TimeSeries} list to cloud monitoring using {@link MetricServiceClient}
+   * @param timeSeriesList {@link List<TimeSeries>} values
+   * @param metricServiceClient {@link MetricServiceClient}
+   * @param projectName Project name string where the metrics will be sent.
+   */
+  public static void send(List<TimeSeries> timeSeriesList, MetricServiceClient metricServiceClient,
+                          String projectName) {
+    ProjectName name = ProjectName.of(projectName);
+    for (List<TimeSeries> partitionedList : Lists.partition(timeSeriesList, 200)) {
+      CreateTimeSeriesRequest request = CreateTimeSeriesRequest.newBuilder()
+        .setName(name.toString())
+        .addAllTimeSeries(partitionedList)
+        .build();
+      metricServiceClient.createTimeSeries(request);
+    }
+  }
+
+  private static long getAggregateValue(Map.Entry<TimeSeriesMetadata, List<Long>> entry,
+                                        TimeSeriesMetadata metadata) {
+    switch (metadata.getAggregation()) {
+      case MEAN:
+        return entry.getValue().stream().reduce(0L, Long::sum) / entry.getValue().size();
```

Comment:
nit: Is it okay to lose accuracy via dividing by int?

Reply:
Changed to library average to avoid errors. Current call will do double calculations and then a cast to long.